queue/retry.go

```diff
@@ -14,7 +14,10 @@ import (
 	"github.com/pkg/errors"
 )
 
-type basicRetryHandler struct {
+// BasicRetryHandler implements the amboy.RetryHandler interface. It provides a
+// simple component that can be attached to an amboy.RetryableQueue to support
+// automatically retrying jobs.
+type BasicRetryHandler struct {
```

Comment:
Does this need to be exported?

Reply:
Yes. Since `NewBasicRetryHandler` is exported, the type should be exported. Before you say it should return an interface, my preference in Go is to [accept interfaces, return structs](https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a), because it will implicitly satisfy the interface.

Reply:
I feel so enlightened, thanks for sharing the post.

Reply:
Also, we should bring this up with the team. We generally return interfaces and maintain impls as unexported.

Reply:
👍 Will bring up in the team meeting or retro.